### ISSUE TITLE
Publish configuration schema file to URL 

### ISSUE THREAD

**Issue by lleadbet** [apollographql/router#3518]
**Describe the solution you'd like**

The router currently supports `router config schema` to export the  current version's schema which can be helpful for autocomplete in various editors. Publishing this schema to a publicly available URL could make setting up an editor much easier without requiring an additional file within source control. 

This would ideally also be versioned so that users could request the config for the version they're running (e.g. router.apollo.dev/schema/v1.26.0)

**Describe alternatives you've considered**

Using the current functionality- which works really well, but not as intuitive as the above.

**Comment by BrynCooke** [comment]
I'd really like us to use https://www.schemastore.org/json/
It would require no setup from users in most IDEs.

However the conversation around this got derailed by versioning support.

I'd really like us to reconsider as it's an easy win with a huge impact on users. We would need to introduce a version field though.

**Comment by lleadbet** [comment]
Yeah, the versioning piece is the big one since our config changes semi-frequently with new products and the like. Self-hosting felt like the happy medium, but if there are ways to get schemastore to work, that'd be the ideal state.

**Comment by BrynCooke** [comment]
Suggest that we just publish the latest and add an optional version field for people running old versions of the router.
